common/nexus/failure.go

```diff
@@ -37,6 +37,21 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+const (
+	// FailureSourceHeaderName is the header used to indicate from where the Nexus failure originated.
+	FailureSourceHeaderName = "Temporal-Nexus-Failure-Source"
+	// FailureSourceWorker indicates the failure originated from outside the server (e.g. bad request or on the Nexus worker).
+	FailureSourceWorker = "worker"
+)
+
+type failureSourceContextKeyType struct{}
+
+var FailureSourceContextKey = failureSourceContextKeyType{}
+
+type FailureSourceContextValue struct {
+	Source string
+}
```

Comment:
Use an atomic value instead, what you have isn't concurrency safe: https://pkg.go.dev/sync/atomic#Value